server/src/main/java/org/elasticsearch/search/aggregations/metrics/ExponentialHistogramState.java

```diff
@@ -117,6 +124,96 @@ public void add(ExponentialHistogram histogram) {
         mergedHistograms.add(histogram);
     }
 
+    /**
+     * Returns the number of scalar values added to this histogram, so the sum
+     * of {@link ExponentialHistogram#valueCount()} for all histograms added.
+     * @return the number of values
+     */
+    public long size() {
+        return histogram().valueCount();
+    }
+
+    /**
+     * Returns the fraction of all points added which are &le; x. Points
+     * that are exactly equal get half credit (i.e. we use the mid-point
+     * rule)
+     *
+     * @param x The cutoff for the cdf.
+     * @return The fraction of all data which is less or equal to x.
+     */
+    public double cdf(double x) {
+        ExponentialHistogram histogram = histogram();
+        long numValuesLess = ExponentialHistogramQuantile.estimateRank(histogram, x, false);
+        long numValuesLessOrEqual = ExponentialHistogramQuantile.estimateRank(histogram, x, true);
+        long numValuesEqual = numValuesLessOrEqual - numValuesLess;
+        // Just like for t-digest, equal values get half credit
+        return (numValuesLess + numValuesEqual / 2.0) / histogram.valueCount();
+    }
+
+    /**
+     * Returns an estimate of a cutoff such that a specified fraction of the data
+     * added to this TDigest would be less than or equal to the cutoff.
+     *
+     * @param q The desired fraction
+     * @return The smallest value x such that cdf(x) &ge; q
+     */
+    public double quantile(double q) {
+        return ExponentialHistogramQuantile.getQuantile(histogram(), q);
+    }
+
+    /**
+     * @return an array of the mean values of the populated histogram buckets with their counts
+     */
+    public Collection<Centroid> centroids() {
+        List<Centroid> centroids = new ArrayList<>();
+        addBucketCentersAsCentroids(centroids, histogram().negativeBuckets().iterator(), -1);
+        // negative buckets are in decreasing order, we want increasing order, therefore reverse
+        Collections.reverse(centroids);
```

Comment:
I'm a bit confused by this.
This had me believe that you start with the lowest values and go up to the highest ones:

https://github.com/elastic/elasticsearch/blob/a0f415d2a0ee75bb299c4ed459f07d1f5b3ef4dd/libs/exponential-histogram/src/main/java/org/elasticsearch/exponentialhistogram/FixedCapacityExponentialHistogram.java#L42-L46

I guess the last sentence in the comment is wrong then? The indices are ascending but the highest index for the negative scale has the lowest value. Did I get that right?